code/modules/reagents/chemistry/reagents/food.dm

```diff
@@ -12,11 +12,13 @@
 	var/diet_flags = DIET_OMNI | DIET_HERB | DIET_CARN
 
 /datum/reagent/consumable/on_mob_life(mob/living/M)
-	if(!(M.mind in SSticker.mode.vampires))
-		if(ishuman(M))
-			var/mob/living/carbon/human/H = M
-			if(H.can_eat(diet_flags))	//Make sure the species has it's dietflag set, otherwise it can't digest any nutrients
-				H.adjust_nutrition(nutriment_factor)	// For hunger and fatness
+	if(M.mind.has_antag_datum(/datum/antagonist/vampire))
+		return ..()
+
+	if(ishuman(M))
```

Comment:
```suggestion
	var/is_vamp = M.mind?.has_antag_datum(/datum/antagonist/vampire)
	if(ishuman(M) && !is_vamp)
```
`M.mind` can be `null` here. This way of writing the code also removes the need to call `return ..()`. More a personal preference of mine btw so don't have to rewrite it like this. But at least be sure to fix the NPE